### PR TITLE
Convert site to Next.js, rewrite docs.

### PR DESCRIPTION
**Pull request by leerob**
In preparation for a new blog post, this PR converts the SWC site from Docusarus to Next.js (using Nextra). It's largely based on the SWR docs site. Once I got started... I also decided to rewrite, clarify, and clean up the entirety of the documentation.

### CODE REVIEW

File: public/favicon/site.webmanifest [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{
+    "name": "",
+    "short_name": "",
+    "icons": [
+        {
+            "src": "/android-chrome-192x192.png",
```
Comment (leerob) [inline public/favicon/site.webmanifest:6]: I also added a new SWC favicon based on the logo 😄

Comment (kdy1) [reply public/favicon/site.webmanifest:6]: 👍 Thanks

**Comment by leerob** [comment]
Temporarily removed the type checking blog post, as we can introduce that in a standalone PR. I believe this is now ready for review!

**Comment by leerob** [comment]
After merging, I can also help land this once we add more details 😄 https://github.com/swc-project/website/pull/119

**Comment by leerob** [comment]
Can likely also archive or remove https://github.com/swc-project/swc-project.github.io

File: pages/docs/usage/plugins.md [hunk 2]
```diff
@@ -0,0 +1,316 @@
+# Plugins
+
+You can create and use custom plugins with SWC.
+
+> Warning: Passing the AST to the JS Plugin system is a performance bottleneck.
+>
+> Therefore, we are concidering two approaches for the next SWC major version:
+>
+> 1. Creating a plugin system in Rust ([#2337](https://github.com/swc-project/swc/discussions/2337))
```
Comment (kdy1) [inline pages/docs/usage/plugins.md:9]: I think https://github.com/swc-project/swc/issues/2635 is better link for this.

File: pages/docs/usage/plugins.md [hunk 3]
```diff
@@ -0,0 +1,316 @@
+# Plugins
```
Comment (kdy1) [inline pages/docs/usage/plugins.md:1]: I think Plugins (v1) would be better.

File: pages/playground/index.md [hunk 4]
```diff
@@ -0,0 +1,18 @@
+---
+title: SWC Playground
+full: true
+---
+
+<iframe
```
Comment (kdy1) [inline pages/playground/index.md:6]: Genious, thanks!

**Review by kdy1 — COMMENTED** [review]
Btw, I love the new sidebar in docs.


I found that `swcpack` is shown twice while searching
image

File: pages/blog/swc-1.mdx [hunk 5]
```diff
@@ -1,23 +1,27 @@
 ---
-title: Introducing swc 1.0
-author: DongYoon Kang
-authorURL: "http://github.com/kdy1"
-authorFBID: 100024888122318
+image: https://user-images.githubusercontent.com/9113740/140682117-5e9f460e-c8f1-447c-88a0-e75f80779260.png
+description: 'SWC (Speedy Web Compiler) is a super-fast JavaScript and TypeScript compiler.'
```
Comment (iheyunfei) [inline pages/blog/swc-1.mdx:3]: I thought this is [spdy web compiler](https://github.com/swc-project)?

Comment (leerob) [reply pages/blog/swc-1.mdx:3]: Good point 😄 I will fix this!

Comment (leerob) [reply pages/blog/swc-1.mdx:3]: Actually, the 1.0 blog posts "(speedy web compiler) "

@kdy1 which do you prefer?

Comment (kdy1) [reply pages/blog/swc-1.mdx:3]: It was `spdy web compiler`, but as many people said it's misleading.
So I cahnged it to `speedy web compiler`.

**Comment by leerob** [comment]
I just realized we probably don't want to remove all of the old changelog files, even if we move to GitHub releases, for historical context. So to keep those around in case people want to read the markdown versions, I readded them to an `old-changelog` folder on GitHub, which we can link to 🛩️

**Comment by kdy1** [comment]
@leerob Thanks!

**Comment by leerob** [comment]
I believe we're good to merge this, if you are ready 😄

**Pull request merged**